src/entity-generator/entity-generator.cpp

```diff
@@ -497,9 +498,13 @@ int main(int argc, char *argv[]) {
     waitForCompletion(headers["Location"], curl);
 
     if (!options.ungoverned) {
-      std::cout << getTimeString() << ": Sleeping for " << options.timeInterval
-                << " seconds" << std::endl;
-      sleep(options.timeInterval);
+      int workTime = nowUTC() - startTime;
```

Comment:
This doesn't really solve the underlying issue.  It does make it less bad, but it doesn't remove it.

The print below in particular is a pretty slow operation that isn't being accounted for in your sleepTime.

But beyond that, sleep isn't guaranteed to wake up the process right away.  And since each iteration tries to add the full interval, you'll still end up falling behind over a long enough time range.

The right way to do this is to compute your target end time for each iteration (based on start time (which needs to be measured before the loop), timeInterval, and count).  Then you determine how long to sleep based on the gap between the current time and the target time.  This has the added benefit of automatically catching up when one iteration runs long.  It's also easy to detect when things are running behind (because the computed sleepTime is negative) so you can notify the user when it's happening.